src/securechip/securechip.h

```diff
@@ -21,6 +21,17 @@
 #include <stddef.h>
 #include <stdint.h>
 
+typedef enum {
+    SC_ERR_ZONE_UNLOCKED_CONFIG = -1,
+    SC_ERR_ZONE_UNLOCKED_DATA = -2,
+    SC_ERR_CONFIG_MISMATCH = -3,
+    SC_ERR_SLOT_UNLOCKED_IO = -4,
+    SC_ERR_SLOT_UNLOCKED_AUTH = -5,
+    SC_ERR_SLOT_UNLOCKED_ENC = -6,
+    SC_ERR_IFS = -7,
+    SC_ERR_FACTORYSETUP = -8,
+} securechip_error_t;
```

Comment:
I like it but would it make sense to have a global src/status.h where we'd define all possible enums, not just the securechip: fatfs, libwally things, nanopb, usb and so on. When handling an API call, we could then catch exact error enum and propagate it to the client, no matter at which call chain it occured. This of course requires that all errors have unique enum number.

Reply:
Maybe not this time some but somewhere in the future.